components/brave_wallet_ui/page/screens/send/components/account-selector/account-selector.tsx

```diff
@@ -75,14 +76,24 @@ export const AccountSelector = (props: Props) => {
       return []
     }
 
-    // TODO(apaymyshev): for bitcoin should allow sending to my account, but
-    // from different keyring (i.e. segwit -> taproot)
-    // https://github.com/brave/brave-browser/issues/29262
-    return accounts.filter(
-      (account) =>
-        account.accountId.coin === selectedNetwork.coin &&
-        account.accountId.keyringId === selectedAccountId.keyringId ||
-        (asset?.contractAddress === "" && isFVMAccount(account)))
+    if (selectedNetwork.coin !== BraveWallet.CoinType.FIL) {
+      // TODO(apaymyshev): for bitcoin should allow sending to my account, but
+      // from different keyring (i.e. segwit -> taproot)
+      // https://github.com/brave/brave-browser/issues/29262
+      return accounts.filter(
+        (account) =>
+          account.accountId.coin === selectedNetwork.coin &&
+          account.accountId.keyringId === selectedAccountId.keyringId ||
+          (asset?.contractAddress === "" && isFVMAccount(account)))
+    } else {
+      let filecoinAccounts = accounts.filter((account) =>
+        (account.accountId.coin === selectedNetwork?.coin &&
+         account.accountId.keyringId === selectedAccountId?.keyringId))
+      let fevmAccounts = accounts.filter((account) =>
+        (selectedAccountId?.coin === CoinType.FIL &&
```

Comment:
superfluous condition?

`selectedAccountId?.coin === CoinType.FIL` is always true regardless of the account variable within the filter.